### PR TITLE
TG-4232 | fixing tests and adding connector update option

### PR DESCRIPTION
**Pull request by romankor**
- JIRA Ticket URL: https://twingate.atlassian.net/browse/TG-4232

## Changes
Allow connector `name` field updates since it is now supported in the public twingate API

### CODE REVIEW

File: twingate/client_connector.go [hunk 1]
```diff
@@ -20,13 +20,20 @@ type Connectors struct {
 
 const connectorResourceName = "connector"
 
-type createConnectorQuery struct {
+type createConnectorMutation struct {
```
Comment (liorr) [inline twingate/client_connector.go:23]: Please roll back this change from query to mutation even though it is correct this is currently not our standard and we need to change it everywhere if we do it, so this is not the appropriate PR for this change.

File: twingate/client_connector.go [hunk 2]
```diff
@@ -54,6 +61,29 @@ func (client *Client) createConnector(remoteNetworkID graphql.ID) (*Connector, e
...
+
+	variables := map[string]interface{}{
+		"connectorId":   connectorID,
+		"connectorName": connectorName,
+	}
+	r := updateConnectorMutation{}
+
+	err := client.GraphqlClient.NamedMutate(context.Background(), "updateConnector", &r, variables)
+	if err != nil {
+		return NewAPIError(err, "update", connectorResourceName)
```
Comment (liorr) [inline twingate/client_connector.go:77]: `return NewAPIErrorWithID(err, "update", connectorResourceName, connectorID)`

File: twingate/client_connector.go [hunk 3]
```diff
@@ -54,6 +61,29 @@ func (client *Client) createConnector(remoteNetworkID graphql.ID) (*Connector, e
...
+	}
+
+	if !r.ConnectorUpdate.Ok {
+		return NewAPIError(NewMutationError(r.ConnectorUpdate.Error), "update", connectorResourceName)
```
Comment (liorr) [inline twingate/client_connector.go:81]: `return NewAPIErrorWithID(NewMutationError(r.ConnectorUpdate.Error), "update", resourceResourceName, resource.ID)`

File: twingate/resource_connector_test.go [hunk 4]
```diff
@@ -21,25 +23,27 @@ func TestAccRemoteConnector_basic(t *testing.T) {
...
+						resource.TestMatchResourceAttr(connectorResource, "name", regexp.MustCompile("tf-acc.*")),
 					),
 				},
 			},
 		})
 	})
 }
 
-func testTwingateConnector(remoteNetworkName string) string {
+func testTwingateConnectorWithCustomName(remoteNetworkName string, connectorName string) string {
```
Comment (liorr) [inline twingate/resource_connector_test.go:37]: let's have two test one for connector with name and one for connector without name

File: twingate/resource_resource_test.go [hunk 5]
```diff
@@ -44,7 +44,6 @@ func TestAccTwingateResource_basic(t *testing.T) {
 				Check: resource.ComposeTestCheckFunc(
 					resource.TestCheckResourceAttr("twingate_remote_network.test", "name", remoteNetworkName),
 					resource.TestCheckResourceAttr("twingate_resource.test", "name", resourceName),
-					resource.TestMatchResourceAttr("twingate_connector.test_1", "name", regexp.MustCompile("tf-acc.*")),
```
Comment (liorr) [inline twingate/resource_resource_test.go:47]: why did we remove this

File: twingate/resource_resource_test.go [hunk 6]
```diff
@@ -44,7 +44,6 @@ func TestAccTwingateResource_basic(t *testing.T) {
...
 					resource.TestMatchResourceAttr("twingate_connector_tokens.test_1", "access_token", regexp.MustCompile(".*")),
```
Comment (liorr) [inline twingate/resource_resource_test.go:47]: why the `_1` here?

**Comment by romankor** [comment]
@liorr i fixed the covergage , please review again

**Review by liorr — APPROVED** [review]
GTM

**Pull request merged**